### PR TITLE
Implement ccm apply store function - Closes #7025

### LINKED ISSUE #7025: Implement apply() function in InteroperabilityStore
### Description

Implement apply function as described in [LIP0049:InternalFuncs:apply(ccu,ccm)](https://github.com/LiskHQ/lips/blob/aced79e3a90c893f25c8f0735ce73732f17cced6/proposals/lip-0049.md#apply) in `BaseInteroperabilityStore` 

- Make sure `ccu`[(CrossChainUpdate)](https://github.com/LiskHQ/lips/blob/aced79e3a90c893f25c8f0735ce73732f17cced6/proposals/lip-0053.md#parameters-schema) and `ccm`[(CrossChainMessage)](https://github.com/LiskHQ/lips/blob/aced79e3a90c893f25c8f0735ce73732f17cced6/proposals/lip-0049.md#cross-chain-message-schema) interface is declared if not then declare it.

- Make sure `terminatedStateAccount` and `sendInternal` is implementation exists
- In order to use `terminatedStateAccount` and `sendInternal`, initialise InteroperabilityStore by passing `getStore` and `interoperableModules`

### Acceptance Criteria

- Should have unit test cases covering all the code paths

### Additional Information

- Should be done after the issue that implements `terminatedStateAccount` and `sendInternal` internal functions

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

This PR resolves #7025 

### How was it solved?

- [🌱 Add types, context and constants for ccm apply](https://github.com/LiskHQ/lisk-sdk/commit/a601ae4502614f43f4507dc09c8e58e46c28a4f1)
- [🌱 Add get ccm size util](https://github.com/LiskHQ/lisk-sdk/commit/9897f224063341bee8cf8b930454b352d3e8138b)
- [🌱 Implement apply ccm](https://github.com/LiskHQ/lisk-sdk/commit/1adfe024c78e4ed469b7165eed725f1cc8b082ab)
- [✅ Add unit test for ccm apply](https://github.com/LiskHQ/lisk-sdk/commit/72783e8899721cd6db44f1a7c010db9d7e6bc3ce)

### How was it tested?

`npm run test:unit test/unit/modules/interoperability/store.spec.ts`

### CODE REVIEW

File: framework/src/testing/create_contexts.ts [hunk 1]
```diff
@@ -273,6 +274,21 @@ const createCCAPIContext = (params: {
...
+	feeAddress: Buffer;
+	stateStore?: StateStore;
+	logger?: Logger;
+	networkIdentifier?: Buffer;
+	getAPIContext?: () => APIContext;
+	eventQueue?: EventQueue;
+}): CCCommandExecuteContext => {
+	const ccAPIContext = createCCAPIContext(params);
+	return {
+		...ccAPIContext,
```
Comment (mitsuaki-u) [inline framework/src/testing/create_contexts.ts:288]: why not just return `ccAPIContext`

File: framework/test/unit/modules/interoperability/store.spec.ts [hunk 2]
```diff
@@ -361,4 +365,224 @@ describe('Base interoperability store', () => {
...
+		it('should return immediately if sending chain is terminated', async () => {
+			// Arrange
+			mainchainStoreLocal.hasTerminatedStateAccount = jest.fn().mockResolvedValue(true);
+
+			// Act & Assert
+			await expect(mainchainStoreLocal.apply(ccmApplyContext)).resolves.toBeUndefined();
+			expect(mod1.crossChainAPI.beforeApplyCCM).toBeCalledTimes(0);
+		});
+
+		it('should return call all the interoperable beforeApplyCCM hooks', async () => {
```
Comment (mitsuaki-u) [inline framework/test/unit/modules/interoperability/store.spec.ts:465]:
```suggestion
		it('should call all the interoperable beforeApplyCCM hooks', async () => {
```

File: framework/test/unit/modules/interoperability/store.spec.ts [hunk 3]
```diff
@@ -361,4 +365,224 @@ describe('Base interoperability store', () => {
...
+				mockGetStore,
+				new Map().set(1, mod),
+			);
+			mainchainStoreLocal.hasTerminatedStateAccount = jest.fn().mockResolvedValue(false);
+			jest.spyOn(mainchainStoreLocal, 'sendInternal');
+
+			// Act & Assert
+			await expect(mainchainStoreLocal.apply(ccmApplyContext)).resolves.toBeUndefined();
+			expect(mod.crossChainAPI.beforeApplyCCM).toBeCalledTimes(1);
+			expect(mod.crossChainAPI.beforeApplyCCM).toBeCalledTimes(1);
```
Comment (mitsuaki-u) [inline framework/test/unit/modules/interoperability/store.spec.ts:490]: Duplicate assertion

File: framework/test/unit/modules/interoperability/store.spec.ts [hunk 4]
```diff
@@ -361,4 +365,224 @@ describe('Base interoperability store', () => {
...
+				mockGetStore,
+				new Map().set(3, mod1),
+			);
+			mainchainStoreLocal.hasTerminatedStateAccount = jest.fn().mockResolvedValue(false);
+			jest.spyOn(mainchainStoreLocal, 'sendInternal');
+
+			// Act & Assert
+			await expect(mainchainStoreLocal.apply(ccmApplyContext)).resolves.toBeUndefined();
+			expect(mod1.crossChainAPI.beforeApplyCCM).toBeCalledTimes(1);
+			expect(mod1.crossChainAPI.beforeApplyCCM).toBeCalledTimes(1);
```
Comment (mitsuaki-u) [inline framework/test/unit/modules/interoperability/store.spec.ts:509]: Same as above

File: framework/test/unit/modules/interoperability/store.spec.ts [hunk 5]
```diff
@@ -361,4 +365,224 @@ describe('Base interoperability store', () => {
...
+				mockGetStore,
+				new Map().set(1, mod),
+			);
+			mainchainStoreLocal.hasTerminatedStateAccount = jest.fn().mockResolvedValue(false);
+			jest.spyOn(mainchainStoreLocal, 'sendInternal');
+
+			// Act & Assert
+			await expect(mainchainStoreLocal.apply(ccmApplyContext)).resolves.toBeUndefined();
+			expect(mod.crossChainAPI.beforeApplyCCM).toBeCalledTimes(1);
+			expect(mod.crossChainAPI.beforeApplyCCM).toBeCalledTimes(1);
```
Comment (mitsuaki-u) [inline framework/test/unit/modules/interoperability/store.spec.ts:539]: Same as above

File: framework/test/unit/modules/interoperability/store.spec.ts [hunk 6]
```diff
@@ -361,4 +365,224 @@ describe('Base interoperability store', () => {
...
+			mainchainStoreLocal.hasTerminatedStateAccount = jest.fn().mockResolvedValue(false);
+			jest.spyOn(mainchainStoreLocal, 'sendInternal');
+			const executeCCMContext = testing.createExecuteCCMsgAPIContext({
+				...beforeSendCCMContext,
+			});
+
+			// Act & Assert
+			await expect(mainchainStoreLocal.apply(ccmApplyContext)).resolves.toBeUndefined();
+			expect(mod.crossChainAPI.beforeApplyCCM).toBeCalledTimes(1);
+			expect(mod.crossChainAPI.beforeApplyCCM).toBeCalledTimes(1);
```
Comment (mitsuaki-u) [inline framework/test/unit/modules/interoperability/store.spec.ts:577]: Same as above

**Review by mitsuaki-u — COMMENTED** [review]
Minor comments but LGTM

**Review by mitsuaki-u — APPROVED** [review]
🎉

**Review by Incede — APPROVED** [review]

**Pull request merged**